jerry-core/ecma/builtin-objects/ecma-builtin-date.cpp

```diff
@@ -38,19 +41,258 @@
  */
 
 /**
+ * Helper function to try to parse a part of a date string
+ *
+ * @return NaN if cannot read from string, ToNumber() otherwise
+ */
+static ecma_number_t
+ecma_date_parse_date_chars (lit_utf8_iterator_t *iter, /**< iterator of the utf8 string */
+                            uint32_t num_of_chars) /**< number of characters to read and convert */
+{
+  ecma_number_t ret_value = ecma_number_make_nan ();
+  lit_utf8_size_t copy_size = 0;
+  const lit_utf8_byte_t *str_start_p = iter->buf_p + iter->buf_pos.offset;
+
+  while (num_of_chars--)
+  {
+    if (lit_utf8_iterator_is_eos (iter))
+    {
+      copy_size = 0;
+      break;
+    }
+
+    copy_size += lit_get_unicode_char_size_by_utf8_first_byte (*iter->buf_p);
+    lit_utf8_iterator_incr (iter);
+  }
+
+  if (copy_size > 0)
+  {
+    ecma_string_t *str_p = ecma_new_ecma_string_from_utf8 (str_start_p, copy_size);
+    ret_value = ecma_string_to_number (str_p);
```

Comment:
What happens if the prefix is number? E.g. parsing 200- as year? You can read 4 characters, but only the first 3 are decimals. What about hex/octal numbers? '0x40' as year? Or exponential stuff, '20e5' as year?